purchase_order_type/models/purchase_order.py

```diff
@@ -24,3 +24,9 @@ def _default_order_type(self):
     def onchange_partner_id_purchase_order_type(self):
         if self.partner_id.purchase_type:
             self.order_type = self.partner_id.purchase_type.id
+
+    @api.onchange('order_type')
+    def onchange_purchase_order_type(self):
+        if self.order_type:
+            self.incoterm_id = self.order_type.incoterm_id.id
```

Comment:
This should be both also conditional. If the value is set, then you overwrite whatever it has. If not, you leave the previous value.